### DISCUSSION TITLE
Using `IncludeAllTimeSeries` API is not consistent with `IncludeTimeSeries`

### DISCUSSION THREAD

**Discussion by kamranayub** [ravendb/ravendb#19344] [General]
Noticed some unexpected behavior with the undocumented `includeBuilder.IncludeAllTimeSeries(...)` API.

Use case: I store dynamic time series on a document and need to migrate/move all the data to a new document so I need to load all of it. There could be 5 or 50 time series, so including "all" would be ideal. This is a one-time migration so it's okay if it's a little slower.

Unlike `IncludeTimeSeries`, I cannot pass `DateTime.MinValue` or `DateTime.MaxValue`, it takes Sparrow time values.

So what I have so far is:

```
_user = await Session.LoadAsync<User>(args.UserId, includeBuilder =>
{
    includeBuilder.IncludeAllTimeSeries(Raven.Client.Documents.Operations.TimeSeries.TimeSeriesRangeType.Last, Sparrow.TimeValue.FromMonths(10 * 12 /* 10 years should be enough */));
});
```

Originally, I tried using `Last` with `Sparrow.TimeValue.MaxValue` but I got:

> InvalidOperationException: System.InvalidOperationException: Got unexpected TimeValueUnit 'None'. Only the following are supported: 'Second' or 'Month'.

So I chose 10 years since I know there's no data older than 2 years in there.

It doesn't seem like there's an equivalent to `IncludeTimeSeries(name, DateTime.MinValue, DateTime.MaxValue)` unless I am missing something with the Sparrow type system.

Note that with setting the 10 years, I needed to pass `from: DateTime.Now.AddMonths(-120)` to prevent extra requests because otherwise `DateTime.MinValue` would cause an extra request sometimes -- not sure why.

I just wanted to bring this to attention since this API is undocumented right now, but I did find it useful for what I wanted.

**Comment by aviv86** [comment]
Hi Kamran
Thanks for reporting this
not sure why there's no API equivalent for including all timesereis by `from` and `to` dates
I created a ticket to add support for this, you can follow it here:
https://issues.hibernatingrhinos.com/issue/RavenDB-22959/Add-support-for-including-all-timeseries-by-from-to-dates

**Comment by riccardominato** [comment]
I have the same problem.

**Comment by kamranayub** [comment]
This has since been fixed btw, so you can do: `.IncludeAllTimeSeries(from: null, to: null)`
